### ISSUE TITLE
while converting some jpg files to crn, they end upside down

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Urcheon#33]
I discovered the bug while toying with _metro_ and _cruz_ Tremulous map (trying to port them to Unvanquished) and I noticed some textures were upsede-down. I first tought it may be a bug in engine in the `crunch` tool, then after some investigations I discovered it is a bug in the `convert` tool by the ImageMagick project.

See https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1838860

You can have a look at the bug there:

[![convert jpg to tga upside down](https://dl.illwieckz.net/b/imagemagick/bugs/convert-jpg-to-tga-upside-down/convert-jpg-to-tga-upside-down.png?rev=2)](https://dl.illwieckz.net/b/imagemagick/bugs/convert-jpg-to-tga-upside-down/convert-jpg-to-tga-upside-down.png)

Image samples that are known to fail can be found there:
https://dl.illwieckz.net/b/imagemagick/bugs/convert-jpg-to-tga-upside-down/convert-jpg-to-tga-upside-down-samples.zip

Basically, the `urcheon` tool converts using `convert` any textures to a tga file before feeding `crunch`, this allows urcheon to convert any image format to `crn` as long as imagemagick groks the format, even if `crunch` itself does not support the given format.

I used the `tga` format as a transient format because in the past I discovered that `crunch` had some trouble with some `png` file. Since `convert` is known to convert properly those `jpg` to `png` then those `png` to `tga`, we may use a `png` transient file to produce the `tga` file, that would make Urcheon doing:

`jpg` → `png` → `tga` → `crn`

This to be sure to workaround all the bugs from any software stack…

What a mess!

**Comment by illwieckz** [comment]
Bad news, doing `jpg` → `png` → `tga` → `crn` is not enough…

In some case the upside-down bug is done in the `png` → `tga` step…

**Comment by illwieckz** [comment]
It looks like that If I strip metadata using `-strip` option an every `convert` steps on this "two convert step" way, the bug is entirely worked around.
